### PR TITLE
Added `warp` cheatcode

### PR DESCRIPTION
**Pull request by Radinyn**

### CODE REVIEW

File: corelib/cheatcodes.cairo [hunk 1]
```diff
@@ -11,6 +11,8 @@ fn roll(address: felt, caller_address: felt) -> () {
     }
 }
 
+extern fn warp(blk_timestamp: felt, target_contract_address: Option::<felt>) -> Result::<(), felt> nopanic;
```
Comment (MaksymilianDemitraszek) [inline corelib/cheatcodes.cairo:14]: target_contract_address shouldn't be optional no more, this is also probably the reason why your code fails

Comment (MaksymilianDemitraszek) [reply corelib/cheatcodes.cairo:14]: If task description was incorrent then sorry, this is my mistake 😞

File: crates/cairo-lang-sierra/src/extensions/modules/cheatcodes.rs [hunk 2]
```diff
@@ -1,20 +1,23 @@
 use crate::define_libfunc_hierarchy;
 use crate::extensions::lib_func::{
-    LibfuncSignature, SignatureSpecializationContext, SierraApChange, OutputVarInfo, ParamSignature, BranchSignature,
+    BranchSignature, LibfuncSignature, OutputVarInfo, ParamSignature, SierraApChange,
+    SignatureSpecializationContext,
+};
+use crate::extensions::{
+    NamedType, NoGenericArgsGenericLibfunc, OutputVarReferenceInfo, SpecializationError,
```
Comment (karol-bisztyga) [inline crates/cairo-lang-sierra/src/extensions/modules/cheatcodes.rs:7]: nit: looks redundant

File: crates/cairo-lang-sierra/src/extensions/modules/cheatcodes.rs [hunk 3]
```diff
@@ -27,19 +30,17 @@ impl NoGenericArgsGenericLibfunc for DeclareLibFunc {
...
-                            // ty: context.get_concrete_type(ClassHashType::id(), &[])?,
-                            ty: felt_ty.clone(),
-                            ref_info: OutputVarReferenceInfo::SameAsParam { param_idx: 0 },
-                        },
-                    ],
+                    vars: vec![OutputVarInfo {
+                        // ty: context.get_concrete_type(ClassHashType::id(), &[])?,
+                        ty: felt_ty.clone(),
+                        ref_info: OutputVarReferenceInfo::SameAsParam { param_idx: 0 },
+                    }],
```
Comment (karol-bisztyga) [inline crates/cairo-lang-sierra/src/extensions/modules/cheatcodes.rs:43]: nit: looks redundant

**Review by karol-bisztyga — APPROVED** [review]
lgtm

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM 🔥 but has to be tested in protostar as well

**Pull request merged**